### PR TITLE
[AzureStack] Azs.Subscriptions.Admin: Test and Custom scripts

### PR DESCRIPTION
**Pull request by rakku-ms**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/StackAdmin/Azs.Subscriptions.Admin/custom/GeneratedCmdlets.txt [hunk 1]
```diff
@@ -0,0 +1,1230 @@
+<#
+.Synopsis
```
Comment (bganapa) [inline src/StackAdmin/Azs.Subscriptions.Admin/custom/GeneratedCmdlets.txt]: why this text file is needed?

File: src/StackAdmin/Azs.Subscriptions.Admin/custom/New-AzsAcquiredPlan.ps1 [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    [Microsoft.Azure.PowerShell.Cmdlets.SubscriptionsAdmin.Category('Runtime')]
+    [System.Management.Automation.SwitchParameter]
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        if ($PSBoundParameters.ContainsKey(('AcquisitionId')))
+        {
+            $PSBoundParameters['PlanAcquisitionId'] = $PSBoundParameters['AcquisitionId']
```
Comment (bganapa) [inline src/StackAdmin/Azs.Subscriptions.Admin/custom/New-AzsAcquiredPlan.ps1]: Can you please add a comment here saying why this customization is needed

File: src/StackAdmin/Azs.Subscriptions.Admin/custom/New-AzsOffer.ps1 [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        if ( $PSCmdlet.ParameterSetName -eq 'CreateExpanded' )
+        {
+            if ($null -ne $PSBoundParameters['Name'])
+            {
+                $PSBoundParameters['PropertiesName'] = $PSBoundParameters['Name']
```
Comment (bganapa) [inline src/StackAdmin/Azs.Subscriptions.Admin/custom/New-AzsOffer.ps1]: here as well, please add a comment

File: src/StackAdmin/Azs.Subscriptions.Admin/custom/New-AzsPlan.ps1 [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    # Use the default credentials for the proxy
+    ${ProxyUseDefaultCredentials}
+)
+
+    process {
+        if ( $PSCmdlet.ParameterSetName -eq 'CreateExpanded' )
+        {
+            if ($null -ne $PSBoundParameters['Name'])
+            {
+                $PSBoundParameters['PropertiesName'] = $PSBoundParameters['Name']
```
Comment (bganapa) [inline src/StackAdmin/Azs.Subscriptions.Admin/custom/New-AzsPlan.ps1]: please add comments for all the customizations why it is needed

File: src/StackAdmin/Azs.Subscriptions.Admin/readme.md [hunk 5]
```diff
@@ -120,7 +120,7 @@ directive:
       verb: Test
       subject: SubscriptionMoveSubscription
     set:
-      subject: MoveSubscription
+      subject: MoveUserSubscription
```
Comment (bganapa) [inline src/StackAdmin/Azs.Subscriptions.Admin/readme.md]: can we add an alias in order to minimize the breaking change?

File: src/StackAdmin/Azs.Subscriptions.Admin/readme.md [hunk 6]
```diff
@@ -217,7 +223,7 @@ directive:
       parameter-name: State
     set:
       default:
-        script: 'Private'
+        script: Echo "Private"
```
Comment (bganapa) [inline src/StackAdmin/Azs.Subscriptions.Admin/readme.md]: can you please add a comment ? why echo?

File: src/StackAdmin/Azs.Subscriptions.Admin/test/Common.ps1 [hunk 7]
```diff
@@ -0,0 +1,27 @@
+$global:SkippedTests = @(
+)
+
+# Multiple tests
+$global:Location = (Get-AzLocation)[0].Name
```
Comment (bganapa) [inline src/StackAdmin/Azs.Subscriptions.Admin/test/Common.ps1]: need to check whether this is .Name or .Lcoation

File: src/StackAdmin/Azs.Subscriptions.Admin/custom/GeneratedCmdlets.txt (hunk 1 above)
Comment (rakku-ms) [reply src/StackAdmin/Azs.Subscriptions.Admin/custom/GeneratedCmdlets.txt]: not needed. this file contains the default generated cmdlets for all the customized ones.

File: src/StackAdmin/Azs.Subscriptions.Admin/readme.md (hunk 5 above)
Comment (rakku-ms) [reply src/StackAdmin/Azs.Subscriptions.Admin/readme.md]: ok

File: src/StackAdmin/Azs.Subscriptions.Admin/readme.md (hunk 6 above)
Comment (rakku-ms) [reply src/StackAdmin/Azs.Subscriptions.Admin/readme.md]: need this parameter value to be a string. just tried now, other option is '"Private"'.
If the default script is set to 'Private', it is trying to invoke Private and throwing Private cmd not found error.

**Pull request merged**